src/App.tsx

```diff
@@ -15,33 +15,105 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-export const App: React.FC = () => (
-  <div className="App">
-    <button type="button">
-      Start
-    </button>
-
-    <button type="button">
-      Sort alphabetically
-    </button>
-
-    <button type="button">
-      Sort by length
-    </button>
-
-    <button type="button">
-      Reverse
-    </button>
-
-    <button type="button">
-      Reset
-    </button>
-
-    <ul className="Goods">
-      <li className="Goods__item">Dumplings</li>
-      <li className="Goods__item">Carrot</li>
-      <li className="Goods__item">Eggs</li>
-      <li className="Goods__item">...</li>
-    </ul>
-  </div>
-);
+enum SortType {
+  none,
+  name,
+  length,
+}
+
+export const App = () => {
+  const [isVisible, setListVisible] = useState(false);
+  const [isSorted, setListSorted] = useState(SortType.none);
+  const [isReversed, setListReversed] = useState(false);
+  const goods = [...goodsFromServer];
+
+  switch (isSorted) {
+    case SortType.name:
+      goods.sort((good1, good2) => good1.localeCompare(good2));
+      break;
+
+    case SortType.length:
+      goods.sort((good1, good2) => good1.length - good2.length);
+      break;
+
+    default:
+      break;
+  }
```

Comment:
I would suggest one more option you could have used: just create handlers and pass them to buttons, and when those button are clicked – perform the action (i.e. sort, reset, show – and so on)